src/datumaro/plugins/missing_annotation_detection.py

```diff
@@ -0,0 +1,149 @@
+    that spatially overlap with the ground truth annotations.
+
+    For example, the follwing example will produce ``[Bbox(1, 1, 1, 1, label=1, attributes={"score": 0.5})]`` as
+    the missing annotations since ``Bbox(0, 0, 1, 1, label=1, attributes={"score": 1.0})`` is overlapped with the
+    ground-truth annotation, ``Bbox(0, 0, 1, 1, label=0)`` (``label_agnostic_matching=True``)
+
+    .. code-block:: python
+
+        ground_truth_annotations = [
+            Bbox(0, 0, 1, 1, label=0),
+            Bbox(1, 0, 1, 1, label=1),
+            Bbox(0, 1, 1, 1, label=2),
+        ]
+        model_predictions = [
+            Bbox(0, 0, 1, 1, label=1, attributes={"score": 1.0}),
+            Bbox(1, 1, 1, 1, label=1, attributes={"score": 0.5}),
+        ]
+
+    Args:
+        extractor: The dataset used to find missing labeled annotations.
+        launcher: The launcher used to generate model predictions from the dataset.
+        batch_size: The size of the batches used during processing.
+        pairwise_dist: The distance metric used to measure the distance between two annotations.
+            Typically, the distance metric is Intersection over Union (IoU), which is bounded between 0 and 1.
+        score_threshold: The minimum score required for an annotation to be considered
+            a candidate for missing annotations.
+        label_agnostic_matching: If set to false, annotations with different labels are not matched
+            to determine their spatial overlap. In the above example, ``label_agnostic_matching=False``
+            will produce ``model_predictions`` as is since ``Bbox(0, 0, 1, 1, label=1, attributes={"score": 1.0})``
+            has different label with ``Bbox(0, 0, 1, 1, label=0)``.
+    """
+
+    def __init__(
+        self,
+        extractor: IDataset,
+        launcher: Launcher,
+        batch_size: int = 1,
+        pairwise_dist: float = 0.75,
+        score_threshold: Optional[float] = None,
+        label_agnostic_matching: bool = True,
+    ):
+        super().__init__(extractor, launcher, batch_size, append_annotation=False)
+        self._score_threshold = score_threshold
+
+        class LabelAgnosticMatcherContext(IMatcherContext):
+            def get_any_label_name(self, ann: Annotation, label_id: int) -> str:
+                return ""
+
+        label_categories: LabelCategories = self.categories()[AnnotationType.label]
+
+        class LabelGnosticMatcherContext(IMatcherContext):
+            def get_any_label_name(self, ann: Annotation, label_id: int) -> str:
+                return label_categories[label_id]
+
+        self._support_matchers = {
+            AnnotationType.bbox: BboxMatcher(
+                pairwise_dist=pairwise_dist,
+                context=LabelAgnosticMatcherContext()
+                if label_agnostic_matching
+                else LabelGnosticMatcherContext(),
+                match_segments=match_segments_more_than_pair,
+            ),
```

Comment:
do you have any plan to match labels and masks further?

Reply:
We can add `Polygon` for instance segmentation task in the future.